folsom/src/main/java/com/spotify/folsom/MemcacheClient.java

```diff
@@ -104,14 +153,25 @@
   CompletionStage<V> get(String key);
 
   /**
-   * Get the value for the provided key, including the CAS value
+   * Get the value for the provided key, including the CAS value and flags
    *
    * @param key First key, must not be null
    * @return A future representing completion of the request, with the value, including the CAS
-   *     value, or null if the value does not exists.
+   *     value and flags, or null if the value does not exists.
    */
   CompletionStage<GetResult<V>> casGet(String key);
 
+  /**
+   * Get the value for the provided key, including the CAS value and flags
+   *
+   * @param key First key, must not be null
+   * @return A future representing completion of the request, with the value, including the CAS
+   *     value and flags, or null if the value does not exists.
+   */
+  default CompletionStage<GetResult<V>> getWithFlags(String key) {
```

Comment:
This is basically a copy of `casGet`, it looks better when we have a method that explicitly says that it will return flags as well.